include/DolibarrClient.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef _DOLIBARR_CLIENT_HPP_
+#define _DOLIBARR_CLIENT_HPP_
+
+#include <HTTPClient.h>
+
+enum class DolibarrClientStatus {
+    CONFIGURING = 0,
+    READY,
+    SENDING,
+    ERROR
+};
+
+class DolibarrClient {
+public:
+    DolibarrClient() = default;
+    ~DolibarrClient() = default;
+
+    /// @brief Configure the 
+    /// @param endpoint base url of the Dolibarr server
+    /// @param endpoint api key
+    /// @return default warehouse of the product
+    DolibarrClientStatus configure(const char* endpoint, const char* key);
+
+    /// @brief Send product tag to the server
+    /// @param barcode barcode of the product
+    /// @param product product id associeted with the barecode
+    /// @param warehouse default warehouse of the product
+    void sendTag(const String barcode, int& product, int& warehouse);
```

Comment:
Use `@param[out]` to indicate that `product` and `warehouse` are results.
```c
/// @param[out] product product id associated with the barcode
/// @param[out] warehouse default warehouse of the product
```

Also there is a typo in "barecode": it's ***bar***code